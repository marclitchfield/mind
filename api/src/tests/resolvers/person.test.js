import { applySetup, verifyMutations } from '../test-setup';
import gql from 'graphql-tag';

describe('Person tests', () => {
  applySetup();

  test('create union', () => {
    return verifyMutations('Person tests', 'create union', {
      mutations: [
        ({space}) => gql`mutation { Person { person1:post_in_space(input:{sourceId:"${space.id}", title: "test person 1"}) { id } } }`,
        ({space}) => gql`mutation { Person { person2:post_in_space(input:{sourceId:"${space.id}", title: "test person 2"}) { id } } }`,
        ({space}) => gql`mutation { Event { event:post_in_space(input:{sourceId:"${space.id}", datetime: "2010-07-17" title: "test person 2"}) { id } } }`,
        ({person1, person2, event}) => gql`mutation { Person { post_union(person1:"${person1.id}" person2:"${person2.id}" eventId:"${event.id}") } }`
      ],
      query: ({space}) => gql`query {
        Space(id:"${space.id}") {
          people {
            title
            unions {
              participants { title }
              event { title }
            }
          }
        }
      }`
    });
  });

  test('create offspring', () => {
    return verifyMutations('Person tests', 'create offspring', {
      mutations: [
        ({space}) => gql`mutation { Person { parent1:post_in_space(input:{sourceId:"${space.id}", title: "test parent 1"}) { id } } }`,
        ({space}) => gql`mutation { Person { parent2:post_in_space(input:{sourceId:"${space.id}", title: "test parent 2"}) { id } } }`,
        ({space}) => gql`mutation { Person { child:post_in_space(input:{sourceId:"${space.id}", title: "test child"}) { id } } }`,        
        ({space}) => gql`mutation { Event { event:post_in_space(input:{sourceId:"${space.id}", datetime: "2010-07-17" title: "test person 2"}) { id } } }`,
        ({parent1, parent2, child, event}) => gql`mutation { Person { post_offspring(parent1:"${parent1.id}" parent2:"${parent2.id}" childId:"${child.id}" eventId:"${event.id}") } }`
      ],
      query: ({space}) => gql`query {
        Space(id:"${space.id}") {
          people {
            title
            offspring {
              parents { title }
              child { title }
              event { title }
            }
            parents { title }
            children { title parents { title }}
          }
        }
      }`
    });
  });


});