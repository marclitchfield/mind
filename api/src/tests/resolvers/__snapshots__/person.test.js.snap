// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person tests create offspring 1`] = `
Object {
  "Space": Object {
    "__typename": "Space",
    "people": Array [
      Object {
        "__typename": "Person",
        "children": Array [],
        "offspring": Array [],
        "parents": Array [
          Object {
            "__typename": "Person",
            "title": "test parent 1",
          },
          Object {
            "__typename": "Person",
            "title": "test parent 2",
          },
        ],
        "title": "test child",
      },
      Object {
        "__typename": "Person",
        "children": Array [
          Object {
            "__typename": "Person",
            "parents": Array [
              Object {
                "__typename": "Person",
                "title": "test parent 1",
              },
              Object {
                "__typename": "Person",
                "title": "test parent 2",
              },
            ],
            "title": "test child",
          },
        ],
        "offspring": Array [
          Object {
            "__typename": "Offspring",
            "child": Object {
              "__typename": "Person",
              "title": "test child",
            },
            "event": Object {
              "__typename": "Event",
              "title": "test person 2",
            },
            "parents": Array [],
          },
        ],
        "parents": Array [],
        "title": "test parent 2",
      },
      Object {
        "__typename": "Person",
        "children": Array [
          Object {
            "__typename": "Person",
            "parents": Array [
              Object {
                "__typename": "Person",
                "title": "test parent 1",
              },
              Object {
                "__typename": "Person",
                "title": "test parent 2",
              },
            ],
            "title": "test child",
          },
        ],
        "offspring": Array [
          Object {
            "__typename": "Offspring",
            "child": Object {
              "__typename": "Person",
              "title": "test child",
            },
            "event": Object {
              "__typename": "Event",
              "title": "test person 2",
            },
            "parents": Array [],
          },
        ],
        "parents": Array [],
        "title": "test parent 1",
      },
    ],
  },
}
`;

exports[`Person tests create union 1`] = `
Object {
  "Space": Object {
    "__typename": "Space",
    "people": Array [
      Object {
        "__typename": "Person",
        "title": "test person 2",
        "unions": Array [
          Object {
            "__typename": "Union",
            "event": Object {
              "__typename": "Event",
              "title": "test person 2",
            },
            "participants": Array [
              Object {
                "__typename": "Person",
                "title": "test person 2",
              },
              Object {
                "__typename": "Person",
                "title": "test person 1",
              },
            ],
          },
        ],
      },
      Object {
        "__typename": "Person",
        "title": "test person 1",
        "unions": Array [
          Object {
            "__typename": "Union",
            "event": Object {
              "__typename": "Event",
              "title": "test person 2",
            },
            "participants": Array [
              Object {
                "__typename": "Person",
                "title": "test person 2",
              },
              Object {
                "__typename": "Person",
                "title": "test person 1",
              },
            ],
          },
        ],
      },
    ],
  },
}
`;
